### PR TITLE
feat: add support to multiple schemas and the new `KFDDistribution` schema

### LINKED ISSUE #96: Create new `KFDDistribution` kind schema
Create new schema definition to generate models for `KFDDistribution` kind

### PR DESCRIPTION
**Pull request by alessiodionisi**
Closes #96.

Based on `example/new-schema-version` branch

### CODE REVIEW

**Comment by nutellinoit** [comment]
We should also change the behaviour with the certManager setting in the ingress module

Instead of 

```yaml
        nginx:
          # the tls section defines how the tls for the ingresses should be managed
          tls:
            # provider can be certManager, secret
            provider: certManager
        certManager:
          # the configuration for the clusterIssuer that will be created
          clusterIssuer:
            # the name of the clusterIssuer
            name: letsencrypt-fury
            # the email used during issuing procedures
            email: example@sighup.io
            # the type of the clusterIssuer, can be only http01, dns01 integration is not yet supported
            type: http01
```

We should swap type with solvers to make the user free to define the configuration

```yaml
        nginx:
          # the tls section defines how the tls for the ingresses should be managed
          tls:
            # provider can be certManager, secret
            provider: certManager
        certManager:
          # the configuration for the clusterIssuer that will be created
          clusterIssuer:
            # the name of the clusterIssuer
            name: letsencrypt-fury
            # the email used during issuing procedures
            email: example@sighup.io
            # solvers definition
            solvers: []
```

**Comment by nutellinoit** [comment]
Opened a related issue: https://github.com/sighupio/fury-distribution/issues/105

File: kfd.yaml
Comment (Al-Pragliola) [inline kfd.yaml:1]: We should update develop to develop-next

```
  eks:
    version: 1.25
    installer: develop
```

File: schemas/private/ekscluster-kfd-v1alpha2.json [hunk 1]
```diff
@@ -814,6 +814,18 @@
...
       "additionalProperties": false,
+      "oneOf": [
+        {
+          "required": [
+            "type"
+          ]
+        },
+        {
+          "required": [
+            "solvers"
```
Comment (Al-Pragliola) [inline schemas/private/ekscluster-kfd-v1alpha2.json:825]: Here we could reference an existing schema ( if it exists )

**Review by Al-Pragliola — CHANGES_REQUESTED** [review]
Given the new defaults folder, we should delete the file furyctl-defaults.yaml

File: kfd.yaml [hunk 2]
```diff
@@ -11,15 +11,18 @@ modules:
...
   monitoring: v2.1.0
   opa: v1.8.0
-  networking: v1.12.2
+  networking: v1.13.0
 kubernetes:
   eks:
-    version: 1.25
-    installer: v2.0.0
+    version: 1.26
+    installer: develop-next
```
Comment (omissis) [inline kfd.yaml]:
```suggestion
    version: 1.25
    installer: develop
```

File: templates/distribution/manifests/networking/kustomization.yaml.tpl [hunk 3]
```diff
@@ -7,7 +7,29 @@ apiVersion: kustomize.config.k8s.io/v1beta1
...
+{{- end }}
+
+{{- if eq .spec.distribution.common.provider.type "none" }}
+{{- if eq .spec.distribution.modules.networking.type "calico" }}
+  - patches/infra-nodes-tigera.yaml
+{{- end }}
+{{- if eq .spec.distribution.modules.networking.type "cilium" }}
+  - patches/infra-nodes-distro-cilium.yaml
+{{- end }}
+{{- end }}
```
Comment (omissis) [inline templates/distribution/manifests/networking/kustomization.yaml.tpl:35]:
```suggestion
{{- end }}

```

File: templates/distribution/manifests/networking/patches/infra-nodes-distro-cilium.yaml.tpl [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+metadata:
+  name: hubble-ui
+  namespace: kube-system
+spec:
+  template:
+    spec:
+      nodeSelector:
+        {{ template "nodeSelector" $ciliumArgs }}
+      tolerations:
+        {{ template "tolerations" $ciliumArgs }}
```
Comment (omissis) [inline templates/distribution/manifests/networking/patches/infra-nodes-distro-cilium.yaml.tpl:45]:
```suggestion
        {{ template "tolerations" $ciliumArgs }}

```

File: templates/distribution/terraform/velero.tf.tpl [hunk 5]
```diff
@@ -15,3 +17,5 @@ module "velero" {
 output "velero_iam_role_arn" {
   value = module.velero.velero_iam_role_arn
 }
+
+{{- end }}
```
Comment (omissis) [inline templates/distribution/terraform/velero.tf.tpl:21]:
```suggestion
{{- end }}

```

**Review by omissis — APPROVED** [review]
I left minor comments, LGTM

**Pull request merged**